library/packages/src/modules/PackageSystem.rb

```diff
@@ -348,11 +349,11 @@ def Installed(package)
       # so we must avoid it.
       # added --whatprovides due to bug #76181
       rpm_command = "/usr/bin/rpm -q --whatprovides #{package.shellescape}"
-      output = SCR.Execute(path(".target.bash_output"), rpm_command)
+      output = Yast::Execute.on_target.stdout(rpm_command.split(" "))
```

Comment:
that split is very bad idea. As it kills shellescape and e.g. "pkg\ a" is then splitted. So please adapt rpm command above to array.

Reply:
and also it will popup if rpm failed, so not found package. That is wrong.

Reply:
Yep, should use on_target! instead

Reply:
Something like:

```ruby
      output = Yast::Execute.stdout.on_target!(rpm_command)
      log.info "Query installed package with '#{rpm_command.join(" ")}' and result #{output}"

      # return Pkg::IsProvided (package);
      !output.empty?
```

Should be good enough, do you have another suggestion?

Reply:
no, you need to specify allowed_exitstatus to avoid exception when rpm is not found as it is common case.

Reply:
But when no found it is empty as force_stdout is used